### PR TITLE
Abstracts for chapters 2, 4, 12, and 14.

### PR DESCRIPTION
**Pull request by bast**
- Rest of chapters hopefully tonight.
- Ready to go.

### CODE REVIEW

File: chapter-14/recipe-01/abstract.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+In this recipe, we will extend the test example of
+[chapter 4, recipe 1](../../chapter-04/recipe-01),
```
Comment (robertodr) [inline chapter-14/recipe-01/abstract.md]: What about changing the reference to the chapter to "Recipe 1 in Chapter 4"?

Comment (bast) [reply chapter-14/recipe-01/abstract.md]: I agree.

File: chapter-02/recipe-03/abstract.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+In this recipe we use CMake to accommodate the conditional compilation of
+source code that depending on the environment.
```
[the remark this answers is not in the document]
Comment (bast) [reply chapter-02/recipe-03/abstract.md]: Thanks! Will fix that when sending the next batch.

[the remark this answers is not in the document]
Comment (bast) [reply chapter-02/recipe-03/abstract.md]: Oh wait. Will fix that now :-)

**Pull request merged**